azure_functions_worker/extension.py

```diff
@@ -0,0 +1,278 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from types import ModuleType
+from typing import Any, Callable, List, Optional
+import sys
+import importlib
+import logging
+import functools
+from .utils.common import is_python_version
+from .utils.wrappers import enable_feature_by
+from .constants import (
+    PYTHON_ISOLATE_WORKER_DEPENDENCIES,
+    PYTHON_ENABLE_WORKER_EXTENSIONS,
+    PYTHON_ENABLE_WORKER_EXTENSIONS_DEFAULT,
+    PYTHON_ENABLE_WORKER_EXTENSIONS_DEFAULT_39
+)
+from .logging import logger, SYSTEM_LOG_PREFIX
+
+
+# Extension Hooks
+FUNC_EXT_POST_FUNCTION_LOAD = "post_function_load"
+FUNC_EXT_PRE_INVOCATION = "pre_invocation"
+FUNC_EXT_POST_INVOCATION = "post_invocation"
+APP_EXT_POST_FUNCTION_LOAD = "post_function_load_app_level"
+APP_EXT_PRE_INVOCATION = "pre_invocation_app_level"
+APP_EXT_POST_INVOCATION = "post_invocation_app_level"
+
+
+class ExtensionManager:
+    """This marks if the ExtensionManager has already proceeded a detection,
+    if so, the sdk will be cached in ._extension_enabled_sdk
+    """
+    _is_sdk_detected: bool = False
+
+    """This is a cache of azure.functions module that supports extension
+    interfaces. If this is None, that mean the sdk does not support extension.
+    """
+    _extension_enabled_sdk: Optional[ModuleType] = None
```

Comment:
the pydoc comment should go below the variable declarations. Not above. Same for the above `_is_sdk_detected` variable

Reply:
Oh dude, thanks for pointing this out.